src/Raven.Server/Documents/ClusterTransactionWaiter.cs

```diff
@@ -42,7 +42,7 @@ public void SetException(string id, long index, Exception e)
             Database.RachisLogIndexNotifications.NotifyListenersAbout(index, e);
             if (_results.TryGetValue(id, out var task))
             {
-                task.SetException(e);
+                task.TrySetException(e);
```

Comment:
Why do we use Try... methods now?

Reply:
I think I realized why is that - we had this issue that we tried to set the result on an already completed task, right?

Reply:
exactly